returnn/util/basic.py

```diff
@@ -790,15 +790,23 @@ def model_epoch_from_filename(filename):
     # We don't have it in the model, though.
     # For now, just parse it from filename.
     # If TF, and symlink, resolve until no symlink anymore (e.g. if we symlinked the best epoch).
+    file_basename = None
     for potential_ext in [".meta", ".pt"]:
         if not os.path.exists(filename + potential_ext):
             continue
         while os.path.exists(filename + potential_ext) and os.path.islink(filename + potential_ext):
             fn_with_ext_ = os.readlink(filename + potential_ext)
             assert fn_with_ext_.endswith(potential_ext), "strange? %s, %s" % (filename, potential_ext)
             filename = fn_with_ext_[: -len(potential_ext)]
+        file_basename = os.path.basename(filename + potential_ext)
         break
-    m = re.match(".*\\.([0-9]+)", filename)
+    if file_basename is None:  # maybe the input filename already has an extension?
+        # noinspection PyBroadException
+        try:
+            file_basename = os.path.basename(filename)
+        except Exception:
+            return None
+    m = re.match(".*\\.([0-9]+)", file_basename)
```

Comment:
I think this is too complicated. Why not just always do `basename` in the end? Why catch the generic `Exception` for `basename`? I think `basename` can newer throw an exception.

```suggestion
    m = re.match(".*\\.([0-9]+)", os.path.basename(filename))
```